pyproject.toml

```diff
@@ -27,39 +27,39 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: PyPy",
 ]
 dependencies = [
-  "agate~=1.7.0",
+  "agate>=1.7.0,<1.8",
   "colorama>=0.3.9,<0.5",
   "isodate>=0.6,<0.7",
-  "jsonschema~=4.0",
+  "jsonschema>=4.0,<5.0",
   "Jinja2~=3.0",
-  "mashumaro[msgpack]~=3.9",
+  "mashumaro[msgpack]>=3.9,<4.0",
   "pathspec>=0.9,<0.12",
   "protobuf>=4.0.0,<5.0.0",
-  "python-dateutil~=2.0",
-  "requests<3.0.0",
-  "typing-extensions~=4.4",
+  "python-dateutil>=2.0,<3.0",
+  "requests<3.0.0",  # needs to match dbt-core
+  "typing-extensions>=4.4,<5.0",
 ]
 
 [project.optional-dependencies]
 lint = [
-    "black~=23.3",
+    "black>=23.3,<24.0",
     "flake8",
     "flake8-pyproject",
     "flake8-docstrings",
-    "mypy~=1.3",
-    "pytest~=7.3",  # needed for linting tests
-    "types-Jinja2~=2.11",
-    "types-jsonschema~=4.17",
-    "types-protobuf~=4.24.0",
-    "types-python-dateutil~=2.8",
-    "types-PyYAML~=6.0",
+    "mypy>=1.3,<2.0",
+    "pytest>=7.3,<8.0",  # needed for linting tests
+    "types-Jinja2>=2.11,<3.0",
+    "types-jsonschema>=4.17,<5.0",
+    "types-protobuf>=4.24,<5.0",
```

Comment:
`>=4.24,<4.25` However, this is very specific. The decision to open this up to the major might be intentional and safe.

Reply:
I opened it up to match protobuf itself.  It didn't start until 4.24 which is why the lower bounds don't match.